lib/AST/Decl.cpp

```diff
@@ -6022,7 +6022,8 @@ ProtocolDecl::ProtocolDecl(DeclContext *DC, SourceLoc ProtocolLoc,
     : NominalTypeDecl(DeclKind::Protocol, DC, Name, NameLoc, Inherited,
                       nullptr),
       ProtocolLoc(ProtocolLoc),
-      PrimaryAssociatedTypeNames(PrimaryAssociatedTypeNames) {
```

Comment:
I think @bnbarham is right that we should keep it on the bridging side, if only to keep big ASTGen PRs like this one from causing changes elsewhere in the compiler.

Reply:
I did this because it felt counterintuitive and fragile to rely on callers to pre-copy, and I forgot to myself. Worst thing we copy twice; I think that’s a better state of affairs than UB, but I won’t insist.

> and we're doing that for inheritedTypes now as well.

I dropped the ball on inherited types because there are a lot more places to adjust in this case and that seemed too much of a detour for this PR. I would be happy to revisit in a follow-up though.

Reply:
I'd say that's a much larger discussion, though probably one we should have. For now as @DougGregor said I'd prefer we we kept large ASTGen implementations like this PR as self contained as they can be.